### PR TITLE
Hmis 956 feature

### PR DESCRIPTION
**Pull request by subrotomukherjee**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/HMIS-956
https://tools.hmcts.net/jira/browse/HMIS-957
https://tools.hmcts.net/jira/browse/HMIS-958

### Change description ###
HMI - Linked Hearing Group end points

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/test/java/uk/gov/hmcts/futurehearings/hmi/unit/testing/testsuites/DELETE_Resources_LinkedHearingGrp_UnitTests.java [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+
+    @Value("${invalidClientSecret}")
+    private String invalidClientSecret;
+
+    private String linkedHearingGroupCtx;
+
+    @BeforeAll
+    void setToken(){
+        accessToken = TestUtilities.getToken(grantType, clientID, clientSecret, tokenURL, scope);
+        linkedHearingGroupCtx = String.format(genericResourcesApiRootContext, "linked-hearing-group/groupClientReference");
```
Comment (Matthew-Summerbell) [inline src/test/java/uk/gov/hmcts/futurehearings/hmi/unit/testing/testsuites/DELETE_Resources_LinkedHearingGrp_UnitTests.java:96]: Should this be a variable?

File: src/test/java/uk/gov/hmcts/futurehearings/hmi/unit/testing/testsuites/POST_Resources_LinkedHearingGrp_UnitTests.java [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+    private String invalidClientSecret;
+
+    private HmiHttpClient httpClient;
+    private String linkedHearingGroupCtx;
+
+    @BeforeAll
+    void setToken(){
+        accessToken = TestUtilities.getToken(grantType, clientID, clientSecret, tokenURL, scope);
+        this.httpClient = new HmiHttpClient(accessToken, targetInstance);
+        linkedHearingGroupCtx = String.format(genericResourcesApiRootContext, "linked-hearing-group");
```
Comment (Matthew-Summerbell) [inline src/test/java/uk/gov/hmcts/futurehearings/hmi/unit/testing/testsuites/POST_Resources_LinkedHearingGrp_UnitTests.java:100]: same

File: src/test/java/uk/gov/hmcts/futurehearings/hmi/unit/testing/testsuites/PUT_Resources_LinkedHearingGrp_UnitTests.java [hunk 3]
```diff
@@ -0,0 +1,273 @@
...
+
+    @Value("${invalidClientSecret}")
+    private String invalidClientSecret;
+
+    private String linkedHearingGroupCtx;
+
+    @BeforeAll
+    void setToken(){
+        accessToken = TestUtilities.getToken(grantType, clientID, clientSecret, tokenURL, scope);
+        linkedHearingGroupCtx = String.format(genericResourcesApiRootContext, "linked-hearing-group/groupClientReference");
```
Comment (Matthew-Summerbell) [inline src/test/java/uk/gov/hmcts/futurehearings/hmi/unit/testing/testsuites/PUT_Resources_LinkedHearingGrp_UnitTests.java:96]: same

File: src/test/java/uk/gov/hmcts/futurehearings/hmi/unit/testing/testsuites/DELETE_Resources_LinkedHearingGrp_UnitTests.java (hunk 1 above)
Comment (subrotomukherjee) [reply src/test/java/uk/gov/hmcts/futurehearings/hmi/unit/testing/testsuites/DELETE_Resources_LinkedHearingGrp_UnitTests.java:96]: genericResourcesApiRootContext is already a variable. "linked-hearing-group/groupClientReference" is a suffix, which may vary for other tests.

File: src/test/java/uk/gov/hmcts/futurehearings/hmi/unit/testing/testsuites/PUT_Resources_LinkedHearingGrp_UnitTests.java (hunk 3 above)
Comment (subrotomukherjee) [reply src/test/java/uk/gov/hmcts/futurehearings/hmi/unit/testing/testsuites/PUT_Resources_LinkedHearingGrp_UnitTests.java:96]: same as above.

**Pull request merged**